src/config.rs

```diff
@@ -38,12 +43,25 @@ pub struct PeerNetConfiguration<T: InitConnectionHandler, M: MessagesHandler> {
     pub peers_categories: PeerNetCategories,
     /// Default category info for all peers not in a specific category (category info, number of connections accepted only for handshake //TODO: Remove when refactored on massa side)
     pub default_category_info: PeerNetCategoryInfo,
+
+    pub public_key: PubKey,
```

Comment:
Because we have PeerNetKeyPair<PubKey> and he need the public key type and if i add  PubKey: PeerNetPubKey in the definition and don't use it, rust is angry and say to me: "you have the PubKey and don't use it so i don't want to compile"

Reply:
You can use a phantom data for this : https://doc.rust-lang.org/std/marker/struct.PhantomData.html